src/cpp/session/SessionMain.cpp

```diff
@@ -1642,6 +1643,12 @@ Error runPreflightScript()
    // always return success
    return Success();
 }
+
+void exitEarly(int status)
+{
+   FileLock::cleanUp();
+   ::exit(status);
```

Comment:
You're right, but shadowing `exit` seems a little dicey to me (what if two competing headers want to shadow `exit`? Unlikely now but could happen somewhere down the road...)